bundler/lib/bundler/ui/json_shell.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Bundler
+  module UI
+    class JsonShell < UI::Shell
+      def table(header, rows, options = {})
+        return unless info?
+        empty_row = Hash[header.keys.map {|k| [k, nil] }]
+        data = rows.map {|row| empty_row.merge(row) }
+
+        require "json"
+        json_text =
+          if options[:pretty] && JSON.respond_to?(:pretty_generate)
+            JSON.pretty_generate(data)
+          else
+            data.to_json
```

Comment:
For parity with pretty

```suggestion
            JSON.generate(data)
```

Reply:
We just dropped support for 2.6 and 2.7 on master. Luxurious!